.github/workflows/maven.yaml

```diff
@@ -25,11 +28,32 @@ jobs:
         with:
           distribution: 'temurin'
           java-version: '11'
-      - name: icu4j and releaseCLDR
+      - name: Store Maven project version
+        id: mvn-proj-version
+        run: |
+          cd icu4j
+          version="`mvn help:evaluate -Dexpression=project.version -q -DforceStdout`"
+          echo "version=$version"  # debug/info logging for our own purposes, before sending to Github
+          echo "version=$version" >> "$GITHUB_OUTPUT"
+      - name: Assess Maven version release/snapshot
+        id: mvn-proj-version-type
+        run: |
+          if echo "${{ steps.mvn-proj-version.outputs.version }}" | grep -E "\-|SNAPSHOT"
```

Comment:
Is that pipe symbol on purpose? It's a snapshot if the version contains either a dash or "SNAPSHOT"?

Reply:
Yeah. I thought `SNAPSHOT` or `-SNAPSHOT` should be enough, but then I thought about some of the nonstandard version numbers used by other artifacts that still use a `-` separator (ex: [`icu4j-for-cldr`](https://github.com/unicode-org/icu/packages/411079)). Those versions aren't "proper" semantic versions, and that artifact isn't a user-facing artifact. So I felt that they should also be caught as not a release => snapshot.

Reply:
sgtm -- Could you please add a comment that explains this? Maybe *something like*
```
# Set version-type=snapshot for tags that contain SNAPSHOT,
# as well as for unicode-org-internal artifacts with a dash like `icu4j-for-cldr`.
```

Reply:
Done.